pkg/webhook/server/certificate/manager_test.go

```diff
@@ -103,7 +103,14 @@ func TestWaitForDeadlineAndRotate(t *testing.T) {
 		},
 	}
 
-	objs := []runtime.Object{mutatingWebhookConfiguration}
+	service := &corev1.Service{
```

Comment:
I'm not sure I understand why we are inserting a service here. can you explain?

Reply:
If cert-manager is the one creating the secret we have to be sure that it's deleted if not needed, best way is to set the service as the owner so whenever the service is removed the secret will be removed as well.

Reply:
ack